files/justfiles/toggles.just

```diff
@@ -197,61 +197,100 @@ toggle-xwayland ACTION="prompt":
+    echo ".bashrc, .bash_profile, and more will be ""${b}""$pending_status""${n}""."
     echo "Do you understand?"
     echo "Please type in \"YES I UNDERSTAND\" and press enter"
-    read ACCEPT
-    if [ "$ACCEPT" == "YES I UNDERSTAND" ]; then
-      if lsattr "${BASH_ENV_FILES[0]}" 2>/dev/null | awk '{print $1}' | grep -q 'i'; then
-        echo "Bash environment '(${BASH_ENV_FILES[@]})' is locked down. Unlocking it."
-        for file in "${BASH_ENV_FILES[@]}"; do
-            run0 chattr -i "$file"
-        done
-      else
-        echo "Bash environment '(${BASH_ENV_FILES[@]})' is unlocked. Locking it."
-        echo "
-    # .bashrc
-
-    # Source global definitions
-    if [ -f /etc/bashrc ]; then
-        . /etc/bashrc
+    if [ "{{ skip_approval }}" == "false" ]; then
+        read -r ACCEPT
+        if [ "$ACCEPT" != "YES I UNDERSTAND" ]; then
+            exit
+        fi
     fi
-
-    # User specific environment
-    if ! [[ "\$PATH" =~ "\$HOME/.local/bin:\$HOME/bin:" ]]; then
-        PATH="\$HOME/.local/bin:\$HOME/bin:\$PATH"
+    echo "Do you want this change to apply to ""${b}""all""${n}"" users (""${user_list[@]}"")? [y/N]"
+    echo "Otherwise, it will only apply to the user who launched this script ($SUDO_USER)."
+    if [ "{{ multiuser }}" == "false" ]; then
+        read -r REPLY
+        # Note this way the default if you press enter is to only effect the user who launched this script
+        if [[ "$REPLY" != [yY]* ]]; then
+    	    user_list=("$SUDO_USER")
+        fi
     fi
-    export PATH
 
-    # Uncomment the following line if you don't like systemctl's auto-paging feature:
-    # export SYSTEMD_PAGER=
+    chattr_operator=$([ "$pending_status" == "locked" ] && echo "+i" || echo "-i")  
+    for user in "${user_list[@]}"; do
+    	echo "Applying for user: $user"
+    	user_home=$(getent passwd "$user" | awk -F':' '{ print $6}')
+    	[[ -d "$user_home" ]] || {echo "Variable \$user_home for $user is somehow empty (check your getent passwd entries); safely exiting."; exit 1;}
 
-    unset rc
-          " > ~/.bashrc
+      BASH_ENV_FILES=(
+        "$user_home/.bashrc"
+        "$user_home/.bash_profile"
+        "$user_home/.config/bash-completion"
+        "$user_home/.profile"
+        "$user_home/.bash_logout"
+        "$user_home/.bash_login"
+      )
 
-        echo "
-    # .bash_profile
-
-    # Get the aliases and functions
-    if [ -f ~/.bashrc ]; then
-        . ~/.bashrc
-    fi
-
-    # User specific environment and startup programs
-        " > ~/.bash_profile
+      BASH_ENV_DIRS=(
+        "$user_home/.bashrc.d/"
+        "$user_home/.config/environment.d/"
+      )
 
+      if [ "$pending_status" == "locked" ]; then
+        current_time=$(date -Iminutes | tr ' ' '_')
         for file in "${BASH_ENV_FILES[@]}"; do
-            run0 chattr +i "$file"
+          cp --preserve=all "$file" "$file_$current_time"
+          rm "$file"
```

Comment:
`rm` will not remove an immutable file; same for directories. Locking already locked files is easy to trigger:

1. Lock users machine-wide.
2. Unlock the current user.
3. Try locking users machine-wide once again.

Protective `chattr -i` would've prevented that.